firmware/Makefile

```diff
@@ -30,7 +30,7 @@ endif
 
 # C++ specific options here (added to USE_OPT).
 ifeq ($(USE_CPPOPT),)
-  USE_CPPOPT = -std=c++11 -fno-rtti -fno-exceptions -fno-use-cxa-atexit -Werror=write-strings -Werror=type-limits
+  USE_CPPOPT = -std=c++17 -Wno-register -fno-rtti -fno-exceptions -fno-use-cxa-atexit -Werror=write-strings -Werror=type-limits
```

Comment:
is this really part of "Improve wall wetting AE to use time constant"? why are we changing this? is this intentional?

Reply:
It's required for the `constexpr float expf_taylor_impl` to work because there's a loop in a `constexpr` function.  I think it actually only requires c++14 to work, but if I'm going to poke it I'm going all the way to 17.